packages/ui/src/components/Tooltip/index.tsx

```diff
@@ -204,42 +226,49 @@ const Tooltip = ({
       // Adding true as third parameter to event listener will detect nested scrolls.
       window.addEventListener('scroll', onScrollDetected, true)
     }
-  }, [
-    visibleInDom,
-    positions.tooltipPosition,
-    generatePositions,
-    placement,
-    text,
-    onScrollDetected,
-  ])
+
+    return () => {
+      window.removeEventListener('scroll', onScrollDetected, true)
```

Comment:
Shouldn't we also clear the timeout here?

Reply:
Which timeout? This one is just a fallback if the tooltip wrapper is being removed from the dom unexpectedly, we don't want to have an opened listener

Reply:
I'm talking about `timer.current`, but not sure if it's necessary or not

Reply:
@QuiiBz we could remove it there can be a case where time is still set and the component is being unmounted, good point